src/software/ai/hl/stp/tactic/move/simulated_move_tactic_test.cpp

```diff
@@ -116,6 +135,105 @@ TEST_F(SimulatedMoveTacticTest, test_autokick_move)
             }
             robotAtPosition(1, world_ptr, destination, 0.05, yield);
             ballKicked(Angle::threeQuarter(), world_ptr, yield);
+            auto stopped_time = world_ptr->getMostRecentTimestamp();
+            while (world_ptr->getMostRecentTimestamp() <
+                   stopped_time + Duration::fromSeconds(3))
+            {
+                yield("Waiting 3 seconds before re-checking conditions");
+            }
+            robotAtPosition(1, world_ptr, destination, 0.05, yield);
+        }};
+
+    std::vector<ValidationFunction> non_terminating_validation_functions = {};
+
+    runTest(terminating_validation_functions, non_terminating_validation_functions,
+            Duration::fromSeconds(10));
+}
+
+TEST_F(SimulatedMoveTacticTest, test_spinning_move_clockwise)
+{
+    Point initial_position = Point(-4, 2);
+    Point destination      = Point(4, 2);
+    setBallState(BallState(Point(1, 1), Vector(0, 0)));
+    addFriendlyRobots({RobotStateWithId{
+        .id          = 1,
+        .robot_state = RobotState(initial_position, Vector(0, 0), Angle::zero(),
+                                  AngularVelocity::quarter())}});
+
+    auto tactic = std::make_shared<MoveTactic>(false);
+    tactic->updateControlParams(destination, Angle::zero(), 0, DribblerMode::OFF,
+                                BallCollisionType::ALLOW, {AutoChipOrKickMode::OFF, 0.0},
+                                MaxAllowedSpeedMode::PHYSICAL_LIMIT, 1.0);
+    setTactic(tactic);
+    setRobotId(1);
+
+    std::vector<ValidationFunction> terminating_validation_functions = {
+        [destination, tactic](std::shared_ptr<World> world_ptr,
+                              ValidationCoroutine::push_type& yield) {
+            robotHasOrientation(1, world_ptr, Angle::half(), Angle::fromDegrees(5),
+                                yield);
+            robotAtPosition(1, world_ptr, destination, 0.05, yield);
+            robotHasOrientation(1, world_ptr, Angle::zero(), Angle::fromDegrees(5),
+                                yield);
+            while (!tactic->done())
+            {
+                yield("Tactic is not done");
+            }
+            auto stopped_time = world_ptr->getMostRecentTimestamp();
+            while (world_ptr->getMostRecentTimestamp() <
+                   stopped_time + Duration::fromSeconds(3))
+            {
+                yield("Waiting 3 seconds before re-checking conditions");
+            }
+            robotAtPosition(1, world_ptr, destination, 0.05, yield);
+            robotHasOrientation(1, world_ptr, Angle::zero(), Angle::fromDegrees(5),
+                                yield);
+        }};
```

Comment:
I don't see any checks that the robot is actually spinning here? Maybe do a simple check like taking a moving average with a 1-2 second time window of robot angular velocity and ensuring it meets the command speed?

Reply:
added check